voter-dapp/src/RetrieveRewards.js

```diff
@@ -154,13 +146,49 @@ function RetrieveRewards({ votingAccount }) {
     }, [roundIds, votingAccount])
   );
 
+  // Get auxillary information about all price requests that were revealed.
+  const priceRequests = useCacheCall(["Voting"], call => {
+    if (!revealedVoteEvents) {
+      return null;
+    }
+
+    const priceRequests = revealedVoteEvents.map(event => {
+      return {
+        identifier: event.returnValues.identifier,
+        time: event.returnValues.time
+      };
+    });
+
+    // Get each price request's status. This includes whether it has been resolved and the last round where it was
+    // voted on.
+    const statuses = call("Voting", "getPriceRequestStatuses", priceRequests);
+
+    if (!statuses) {
+      return null;
+    }
+
+    // Pulls down the price for every resolved request. If the request wasn't resolved, the resolved price isn't
+    // queried and the field is left undefined.
+    let done = true;
+    for (let i = 0; i < priceRequests.length; i++) {
+      const priceRequest = priceRequests[i];
+      const status = statuses[i];
+      priceRequest.lastRound = status.lastVotingRound;
+      if (status.status === PriceRequestStatusEnum.RESOLVED) {
+        priceRequest.resolvedPrice = call("Voting", "getPrice", priceRequest.identifier, priceRequest.time, {
+          from: governorAddress
```

Comment:
question: The `getPrice` function is a public view function. Why do you need this `from: governorAddress` here?

Reply:
It can only be called by approved financial contracts. It will revert if called by the user's EOA. But for `.calls`, you can just tell the node to simulate the call as if it were called from a different address.

I'll add a comment to explain.

Reply:
Added a comment. PTAL.